interactions/models/internal/tasks/triggers.py

```diff
@@ -123,8 +130,11 @@ def __or__(self, other: "BaseTrigger") -> "OrTrigger":
         self.triggers.append(other)
         return self
 
+    def _set_current_trigger(self) -> BaseTrigger | None:
+        self.current_trigger = self.triggers[0] if len(self.triggers) == 1 else min(self.triggers, key=self._get_delta)
+        return self.current_trigger
+
     def next_fire(self) -> datetime | None:
-        if len(self.triggers) == 1:
-            return self.triggers[0].next_fire()
-        trigger = min(self.triggers, key=self._get_delta)
-        return trigger.next_fire()
+        if not self._set_current_trigger():
+            return None
+        return self.current_trigger.next_fire()
```

Comment:
Very nitpicky suggestion 

```suggestion
        return self.current_trigger.next_fire() if self._set_current_trigger() else None
```